rl/approximate_dynamic_programming.py

```diff
@@ -178,12 +175,9 @@ def return_(s_r: Tuple[S, float], i=i) -> float:
             return r + γ * (v[i-1].evaluate([s]).item() if i > 0 else 0.)
 
         v.append(
-            FunctionApprox.converged(
-                sgd(
-                    approx0,
-                    repeat([(s, mrp.transition_reward(s).expectation(return_))
-                            for s in mu.sample_n(num_state_samples)])
-                ),
+            approx0.solve(
+                [(s, mrp.transition_reward(s).expectation(return_))
```

Comment:
Oh, I really like how `mrp.transition_reward(s).expectation(return_)` reads :). I didn't have this in mind when designing the pieces, and it's fun to see patterns like this emerge from simple foundations.